### PR TITLE
sqlite3_exec replaced by non serializing type aware osquery_exec

### PR DESCRIPTION
**Pull request by guliashvili**
Enables #4828  #4799 
sqlite3_exec serializes everything into the string and does not give the information about the type. shell_exec serializes everything into the string but gives the original type information, however it's written in a low level C code.

osquery_exec does not lose information about the types, does no serialize things into the string(saving time) and is written in a more c++ like code. (Getting rid of callback mess and mallocs)

Currently osquery_exec results are still just serialized into the string, however #4828 we plan to use it soon.  

Please, let me know your thoughts about the improvements.

### CODE REVIEW

File: osquery/sql/sqlite_util.cpp [hunk 1]
```diff
@@ -459,40 +461,85 @@ Status QueryPlanner::applyTypes(TableColumns& columns) {
   return Status(0);
 }
 
-int queryDataCallback(void* argument, int argc, char* argv[], char* column[]) {
-  if (argument == nullptr) {
-    VLOG(1) << "Query execution failed: received a bad callback argument";
-    return SQLITE_MISUSE;
+Status osquery_exec(
```
Comment (j-calvert) [inline osquery/sql/sqlite_util.cpp]: sqlite3_exec handles multi-statement queries.  Unless I'm mistaken, this does not.

Comment (guliashvili) [reply osquery/sql/sqlite_util.cpp]: Thanks for the feedback. Yes that was my intentional decision, this sqlite3_exec was not used for multiple query execution and also should not. When several similar tables are appended union is used, when tables are not similar it means we are parsing and sending a lot of null entries. Thats why i decided to limit this capability in the new function.

Comment (j-calvert) [reply osquery/sql/sqlite_util.cpp]: How can one know that it was never used for multi-statement queries?  In fact, (again, unless I'm mistaken) we have multi-statement queries in our own internal use of osquery.

**Comment by guliashvili** [comment]
as discussed, closing this PR in favor of the #4904, which will use this PR as the reference point for api and variant use.

**Pull request closed without merging**